pkg/controller/gcppubsub/resources/receive_adapter.go

```diff
@@ -23,29 +23,29 @@ import (
 	"k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 // ReceiveAdapterArgs are the arguments needed to create a GCP PubSub Source Receive Adapter. Every
 // field is required.
 type ReceiveAdapterArgs struct {
 	Image          string
-	Scheme         *runtime.Scheme
 	Source         *v1alpha1.GcpPubSubSource
 	Labels         map[string]string
 	SubscriptionID string
 	SinkURI        string
 }
 
+const (
+	credsVolume    = "google-cloud-key"
+	credsMountPath = "/var/secrets/google"
+)
+
 // MakeReceiveAdapter generates (but does not insert into K8s) the Receive Adapter Deployment for
 // GCP PubSub Sources.
-func MakeReceiveAdapter(args ReceiveAdapterArgs) (*v1.Deployment, error) {
-	credsVolume := "google-cloud-key"
-	credsMountPath := "/var/secrets/google"
+func MakeReceiveAdapter(args *ReceiveAdapterArgs) *v1.Deployment {
```

Comment:
:+1: Consolidating error handling in the caller seems like a readability improvement, and makes this function more testable.

Reply:
I actually disagree. I dislike this refactor because it makes this function not create a usable `Deployment`. The caller has the add the owner reference themselves (which is not documented). If they don't, then I think the `Deployment` will leak after the `GcpPubSubSource` is deleted. 

It also opens the potential for a future bug (although very unlikely) where the source passed through args (line 166) is different than the one added as an owner (line 171).